tests/core/test_db_dml_pagination.py

```diff
@@ -43,16 +45,46 @@
     ],
 )
 def test_find_paginated(
-    prefetched: Optional[int], pagination_v_collection: AstraDBCollection
+    prefetched: Optional[int],
+    pagination_v_collection: AstraDBCollection,
+    caplog: pytest.LogCaptureFixture,
 ) -> None:
     options = {"limit": FIND_LIMIT}
     projection = {"$vector": 0}
+    caplog.set_level(logging.INFO)
 
-    paginated_documents_gen = pagination_v_collection.paginated_find(
+    paginated_documents_it = pagination_v_collection.paginated_find(
         projection=projection, options=options, prefetched=prefetched
     )
-    paginated_documents = list(paginated_documents_gen)
+
+    time.sleep(1)
+    if prefetched:
+        # If prefetched is set requests are performed eagerly
+        assert caplog.text.count("HTTP Request: POST") == 3
+    else:
+        assert caplog.text.count("HTTP Request: POST") == 0
+
+    paginated_documents = list(paginated_documents_it)
     paginated_ids = [doc["_id"] for doc in paginated_documents]
     assert all(["$vector" not in doc for doc in paginated_documents])
     assert len(paginated_ids) == FIND_LIMIT
     assert len(paginated_ids) == len(set(paginated_ids))
+
+
+def test_prefetched_thread_terminated(
```

Comment:
would it be ok to add an async one in the async twin test module? (perhaps does not add much, but...)

```
import asyncio
```

and later 

```
async def test_prefetched_thread_terminated(
    async_pagination_v_collection: AstraDBCollection,
) -> None:
    options = {"limit": FIND_LIMIT}
    projection = {"$vector": 0}

    paginated_documents_agen = async_pagination_v_collection.paginated_find(
        projection=projection, options=options, prefetched=PREFETCHED
    )

    assert await anext(paginated_documents_agen) is not None
    del paginated_documents_agen
```

Reply:
The async counterpart doesn't spawn a thread.
It creates an async task which arguably has the same default of leaking.
This can be considered in another PR ?

Reply:
Of course!